enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,358 @@
+OVN Kubernetes to connect the networks of the two CNI network plugins. So that
+pods on each side can still talk to the pods on the other side.
+
+## Motivation
+
+For some Openshift users, they have very high requirement on service
+availability. The current SDN migration solution, that will cause a service
+interruption, is not acceptable.
+
+### Goals
+
+- Migrate the cluster network provider from OpenShift SDN to OVN Kubernetes for a 
+  existing cluster.
+- This's in-place migration without requiring extra nodes.
+- The impact to workload of the migration shall be similar as an OCP upgrade.
+- The solution can work in scale, e.g. in a large cluster with hundreds of
+  nodes.
+- The migration operation shall be able to be rolled back if needed.
+
+### Non-Goals
+
+- Support for migration to other network providers
+- The necessary GUI change in Openshift Cluster Manager.
+- The live migration of the egress IP and firewall configuration.
+
+## Pre-requisites
+
+- This solution relies on the ovnkube hybrid overlay feature. This bug 
+  https://bugzilla.redhat.com/show_bug.cgi?id=2040779 needs to be fixed.
+
+## Proposal
+
+The key problem of do a live SDN migration is that, we need to the connectivity 
+of the cluster network during the migration, when pods are attached to different
+networks. We propose to utilize the OVN Kubernetes hybrid overlay feature to
+connect the networks owned by OpenShift SDN and OVN Kubernetes.
+
+- We will run different plugins on different nodes, but both plugins will know
+  how to reach pods owned by the other plugin, so all pods/services/etc remain
+  connected.
+- During migration, CNO will take origin-plugin nodes one by one and convert them
+  to destination-plugin nodes, rebooting them in the process.
+- The cluster network CIDR will not change. In fact, none of the node host
+  subnets will change.
+- NetworkPolicy will work correctly throughout the migration.
+
+### Limitations
+
+- Multicast is not handled by this proposal.
+- Migration from SDN Multitenant mode is not handled by this proposal.
+
+### User Stories
+
+The service delivery (SD) team (managed OpenShift services ARO, OSD, ROSA) has a
+unique set of requirements around downtime, node reboots, and high degree of
+automation. Specifically, SD need a way to migrate their managed fleet in a way
+that is no more impactful to the customer's workloads than an OCP upgrade, and
+that can be done at scale, in a safe, automated way that can be made
+self-service and not require SD to negotiate maintenance windows with customers.
+The current migration solution needs to be revisited to support these
+(relatively) more stringent requirements.
+
+### Risks and Mitigations
+
+## Design Details
+
+The existing ovn-kubernetes hybrid overlay feature was developed for hybrid
+Windows/Linux clusters. Each ovn-kubernetes node manages an external-to-OVN OVS
+bridge, named br-ext, which acts as the VXLAN source and endpoint for packets
```

Comment:
Is there an overhead / performance impact of using VXLAN and therefore an impact to workloads on the cluster while migration is in progress?

Reply:
if so, is it temp? and if so, how long will the impact last?